lib/improver/tests/ensemble_copula_coupling/ensemble_copula_coupling/test_FromMeanAndVariance.py

```diff
@@ -0,0 +1,127 @@
+        """Test for a valid distribution with shape parameters."""
+        plugin = Plugin(distribution="truncnorm", shape_parameters=[0, np.inf])
+        self.assertEqual(plugin.distribution, stats.truncnorm)
+        self.assertEqual(plugin.shape_parameters, [0, np.inf])
+
+    def test_invalid_distribution(self):
+        """Test for an invalid distribution."""
+        msg = "The distribution requested"
+        with self.assertRaisesRegex(AttributeError, msg):
+            Plugin(distribution="elephant")
+
+
+class Test__repr__(IrisTest):
+
+    """Test string representation of plugin."""
+
+    def test_basic(self):
+        """Test string representation"""
+        expected_string = ("<FromMeanAndVariance: "
+                           "distribution: norm; shape_parameters: []>")
+        result = str(Plugin())
+        self.assertEqual(result, expected_string)
+
+
+class Test__rescale_shape_parameters(IrisTest):
+
+    """Test the _rescale_shape_parameters"""
+
+    def setUp(self):
+        """Set up values for testing."""
+        self.mean = np.array([-1, 0, 1])
+        self.std = np.array([1, 1.5, 2])
+
+    def test_truncated_at_zero(self):
+        """Test scaling shape parameters implying a truncation at zero."""
+        expected = [np.array([1., 0, -0.5]),
+                    np.array([np.inf, np.inf, np.inf])]
+        shape_parameters = [0, np.inf]
+        plugin = Plugin(distribution="truncnorm",
+                        shape_parameters=shape_parameters)
+        plugin._rescale_shape_parameters(self.mean, self.std)
+        self.assertArrayAlmostEqual(plugin.shape_parameters, expected)
+
+    def test_discrete_shape_parameters(self):
+        """Test scaling discrete shape parameters."""
+        expected = [np.array([-3, -2.666667, -2.5]), np.array([7, 4, 2.5])]
+        shape_parameters = [-4, 6]
+        plugin = Plugin(distribution="truncnorm",
+                        shape_parameters=shape_parameters)
+        plugin._rescale_shape_parameters(self.mean, self.std)
+        self.assertArrayAlmostEqual(plugin.shape_parameters, expected)
+
+    def test_alternative_distribution(self):
+        """Test specifying a distribution other than truncated normal. In
+        this instance, no rescaling is applied."""
+        shape_parameters = [0, np.inf]
+        plugin = Plugin(distribution="norm",
+                        shape_parameters=shape_parameters)
+        plugin._rescale_shape_parameters(self.mean, self.std)
+        self.assertArrayEqual(plugin.shape_parameters, shape_parameters)
+
+    def test_no_shape_parameters_exception(self):
+        """Test raising an exception when shape parameters are not specified
+        for the truncated normal distribution."""
+        plugin = Plugin(distribution="truncnorm")
+        msg = "For the truncated normal distribution"
+        with self.assertRaisesRegex(ValueError, msg):
+            plugin._rescale_shape_parameters(self.mean, self.std)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Access to a protected member _rescale_shape_parameters of a client class](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4540588)